Here is the most recent failure, which I can't explain:

```
[ERROR] io.quarkus.arquillian.test.SimpleWarTest  Time elapsed: 0.862 s  <<< ERROR!
org.jboss.arquillian.container.spi.client.container.DeploymentException: Unable to start the runtime runner
	at io.quarkus.arquillian.QuarkusDeployableContainer.deploy(QuarkusDeployableContainer.java:196)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController$3.call(ContainerDeployController.java:151)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController$3.call(ContainerDeployController.java:118)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.executeOperation(ContainerDeployController.java:239)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.deploy(ContainerDeployController.java:118)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.container.impl.client.ContainerDeploymentContextHandler.createDeploymentContext(ContainerDeploymentContextHandler.java:71)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.container.impl.client.ContainerDeploymentContextHandler.createContainerContext(ContainerDeploymentContextHandler.java:54)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.container.impl.client.container.DeploymentExceptionHandler.verifyExpectedExceptionDuringDeploy(DeploymentExceptionHandler.java:47)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController$1.perform(ContainerDeployController.java:92)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController$1.perform(ContainerDeployController.java:77)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.forEachDeployment(ContainerDeployController.java:232)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.forEachManagedDeployment(ContainerDeployController.java:212)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.deployManaged(ContainerDeployController.java:77)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.execute(ContainerEventController.java:96)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:83)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.beforeClass(EventTestRunnerAdaptor.java:89)
	at org.jboss.arquillian.junit.Arquillian$2.evaluate(Arquillian.java:163)
	at org.jboss.arquillian.junit.Arquillian.multiExecute(Arquillian.java:350)
	at org.jboss.arquillian.junit.Arquillian.access$200(Arquillian.java:54)
	at org.jboss.arquillian.junit.Arquillian$3.evaluate(Arquillian.java:177)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.jboss.arquillian.junit.Arquillian.run(Arquillian.java:115)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:383)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:344)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:125)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:417)
Caused by: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.ArcAnnotationProcessor#build threw an exception: javax.enterprise.inject.spi.DeploymentException: javax.enterprise.inject.AmbiguousResolutionException: Ambiguous dependencies for type io.quarkus.arquillian.test.Foo and qualifiers [@Default]
	- java member: io.quarkus.arquillian.test.SimpleClass#foo
	- declared on CLASS bean [types=[io.quarkus.arquillian.test.SimpleClass, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.arquillian.test.SimpleClass]
	- available beans:
		- CLASS bean [types=[io.quarkus.arquillian.test.Foo, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.arquillian.test.Foo]
		- CLASS bean [types=[io.quarkus.arquillian.test.Foo, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.arquillian.test.Foo]
	at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:145)
	at io.quarkus.arquillian.QuarkusDeployableContainer.deploy(QuarkusDeployableContainer.java:188)
	... 89 more
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.ArcAnnotationProcessor#build threw an exception: javax.enterprise.inject.spi.DeploymentException: javax.enterprise.inject.AmbiguousResolutionException: Ambiguous dependencies for type io.quarkus.arquillian.test.Foo and qualifiers [@Default]
	- java member: io.quarkus.arquillian.test.SimpleClass#foo
	- declared on CLASS bean [types=[io.quarkus.arquillian.test.SimpleClass, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.arquillian.test.SimpleClass]
	- available beans:
		- CLASS bean [types=[io.quarkus.arquillian.test.Foo, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.arquillian.test.Foo]
		- CLASS bean [types=[io.quarkus.arquillian.test.Foo, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.arquillian.test.Foo]
	at io.quarkus.builder.Execution.run(Execution.java:124)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:137)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:113)
	at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:110)
	... 90 more
Caused by: javax.enterprise.inject.spi.DeploymentException: javax.enterprise.inject.AmbiguousResolutionException: Ambiguous dependencies for type io.quarkus.arquillian.test.Foo and qualifiers [@Default]
	- java member: io.quarkus.arquillian.test.SimpleClass#foo
	- declared on CLASS bean [types=[io.quarkus.arquillian.test.SimpleClass, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.arquillian.test.SimpleClass]
	- available beans:
		- CLASS bean [types=[io.quarkus.arquillian.test.Foo, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.arquillian.test.Foo]
		- CLASS bean [types=[io.quarkus.arquillian.test.Foo, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.arquillian.test.Foo]
	at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:804)
	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:294)
	at io.quarkus.arc.processor.BeanProcessor.process(BeanProcessor.java:156)
	at io.quarkus.arc.deployment.ArcAnnotationProcessor.build(ArcAnnotationProcessor.java:282)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$1.execute(ExtensionLoader.java:507)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:414)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1538)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1429)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
Caused by: javax.enterprise.inject.AmbiguousResolutionException: Ambiguous dependencies for type io.quarkus.arquillian.test.Foo and qualifiers [@Default]
	- java member: io.quarkus.arquillian.test.SimpleClass#foo
	- declared on CLASS bean [types=[io.quarkus.arquillian.test.SimpleClass, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.arquillian.test.SimpleClass]
	- available beans:
		- CLASS bean [types=[io.quarkus.arquillian.test.Foo, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.arquillian.test.Foo]
		- CLASS bean [types=[io.quarkus.arquillian.test.Foo, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.arquillian.test.Foo]
	at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:393)
	at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:366)
	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:286)
	... 14 more
```

--- a/core/deployment/src/main/java/io/quarkus/runner/RuntimeRunner.java
+++ b/core/deployment/src/main/java/io/quarkus/runner/RuntimeRunner.java
@@ -196,7 +196,7 @@ public class RuntimeRunner implements Runnable, Closeable {
             return this;
         }
 
-        public Builder addAdditionalArchives(Collection<Path> additionalArchive) {
+        public Builder addAdditionalArchives(Collection<Path> additionalArchives) {
             this.additionalArchives.addAll(additionalArchives);
             return this;
         }

--- a/test-framework/arquillian/src/main/java/io/quarkus/arquillian/QuarkusDeployableContainer.java
+++ b/test-framework/arquillian/src/main/java/io/quarkus/arquillian/QuarkusDeployableContainer.java
@@ -182,7 +182,6 @@ public class QuarkusDeployableContainer implements DeployableContainer<QuarkusCo
                     .setTarget(appLocation)
                     .setFrameworkClassesPath(testClassesLocation)
                     .addChainCustomizers(customizers)
-                    .addAdditionalArchives(libraries)
                     .build();
 
             runner.run();

--- a/war-launcher/runner/src/main/java/io/quarkus/legacy/Main.java
+++ b/war-launcher/runner/src/main/java/io/quarkus/legacy/Main.java
@@ -96,11 +96,9 @@ public class Main {
             }
 
             List<URL> urls = new ArrayList<>();
-            List<Path> paths = new ArrayList<>();
             urls.add(archive.toUri().toURL());
             for (Path l : libraries) {
                 urls.add(l.toUri().toURL());
-                paths.add(l);
             }
 
             URLClassLoader ucl = new URLClassLoader(urls.toArray(new URL[urls.size()]), Main.class.getClassLoader());
@@ -110,7 +108,6 @@ public class Main {
                     .setClassLoader(ucl).setTarget(archive)
                     .setFrameworkClassesPath(archive)
                     .setTransformerCache(null)
-                    .addAdditionalArchives(paths)
                     .build()) {
                 runner.run();
             } finally {

